src/mobu/services/business/sia.py

```diff
@@ -0,0 +1,163 @@
+from ...models.business.sia import (
+    SIA2SearchParameters,
+    SIABusinessData,
+    SIABusinessOptions,
+)
+from ...models.user import AuthenticatedUser
+from ...sentry import capturing_start_span, start_transaction
+from .base import Business
+
+T = TypeVar("T", bound="SIABusinessOptions")
+
+__all__ = ["SIABusiness"]
+
+
+class SIABusiness(Business, Generic[T], metaclass=ABCMeta):
+    """Base class for business that executes SIA query.
+
+    This class modifies the core `~mobu.business.base.Business` loop by
+    providing `startup`, `execute`, and `shutdown` methods that know how to
+    execute SIA queries. Subclasses must override `get_next_query` to return
+    the next query they want to execute.
+
+    Parameters
+    ----------
+    options
+        Configuration options for the business.
+    user
+        User with their authentication token to use to run the business.
+    events
+        Event publishers.
+    logger
+        Logger to use to report the results of business.
+    """
+
+    def __init__(
+        self,
+        *,
+        options: T,
+        user: AuthenticatedUser,
+        events: Events,
+        logger: BoundLogger,
+        flock: str | None,
+    ) -> None:
+        super().__init__(
+            options=options,
+            user=user,
+            events=events,
+            logger=logger,
+            flock=flock,
+        )
+        self._running_query: SIA2SearchParameters | None = None
+        self._client: pyvo.dal.SIA2Service | None = None
+        self._pool = ThreadPoolExecutor(max_workers=1)
+        self.query_set: str = self.options.query_set
+
+    async def startup(self) -> None:
+        self._client = self._make_client(self.user.token)
+
+    @abstractmethod
+    def get_next_query(self) -> SIA2SearchParameters:
+        """Get the next SIA query to run.
+
+        Returns
+        -------
+        SIA2SearchParameters
+            SIA query as an SIA2SearchParameters object.
+        """
+
+    async def execute(self) -> None:
+        with start_transaction(
+            name=f"{self.name} - execute",
+            op="mobu.sia.search",
+        ):
+            query = self.get_next_query()
+            with capturing_start_span(op="mobu.sia.search") as span:
+                set_context(
+                    "query_info",
+                    {"query": str(query), "started_at": span.start_timestamp},
+                )
+                self._running_query = query
+
+                success = False
+                try:
+                    if not self._client:
+                        raise RuntimeError("SIABusiness startup never ran")
+                    self.logger.info(f"Running SIA query: {query}")
+                    loop = asyncio.get_event_loop()
+                    await loop.run_in_executor(
+                        self._pool,
+                        self._client.search,
+                        query.to_pyvo_sia2_params(),
+                    )
```

Comment:
Is there a way to execute these async, instead of using a threadpool?  I assume no, just double checking

Reply:
If we use pyvo's SIA search which isn't `async`, that's probably the best we can do for now unless they change it in the future. We could build an asynchronous SIA querying class ourselves that uses something like `aiohttp` but might be a 3-4 days worth of effort kind of thing so I'm not sure if we want to do it now or as an improvement later.

Reply:
Agreed, not worth the effort now